src/polyfill/RTCPeerConnection.ts

```diff
@@ -341,160 +412,191 @@ export default class RTCPeerConnection extends EventTarget implements globalThis
             this.#dataChannelsClosed++;
         });
 
+        for (const transceiver of this.#transceivers) {
+            transceiver.close()
+        }
+        for (const track of this.#tracks) {
+            track.close()
+        }
+
         this.#peerConnection.close();
     }
 
-    createAnswer(): Promise<globalThis.RTCSessionDescriptionInit | any> {
-        return this.#localAnswer;
+    get maxMessageSize (): number {
+        return this.#peerConnection.maxMessageSize()
+    }
+    
+    get maxChannels (): number {
+        return this.#peerConnection.maxDataChannelId()
     }
 
+    createAnswer(): Promise<globalThis.RTCSessionDescriptionInit> & Promise<void> {
+        // @ts-expect-error dont support deprecated overload
+        return this.#localAnswer;
+    }
 
-    createDataChannel(label, opts = {}): RTCDataChannel {
-        const channel = this.#peerConnection.createDataChannel(label, opts);
-        const dataChannel = new RTCDataChannel(channel, opts);
-
+    #handleDataChannel (channel: DataChannel, opts?: DataChannelInitConfig): RTCDataChannel {
+        const dataChannel = new RTCDataChannel(channel, opts, this)
+    
         // ensure we can close all channels when shutting down
-        this.#dataChannels.add(dataChannel);
+        this.#dataChannels.add(dataChannel)
         dataChannel.addEventListener('close', () => {
-            this.#dataChannels.delete(dataChannel);
-            this.#dataChannelsClosed++;
-        });
-
-        return dataChannel;
+            this.#dataChannels.delete(dataChannel)
+        })
+    
+        return dataChannel
+    }
+
+
+    createDataChannel (label: string, opts: globalThis.RTCDataChannelInit = {}): RTCDataChannel {
+        const conf: DataChannelInitConfig = opts
+        if (opts.ordered === false) conf.unordered = true
+        const channel = this.#peerConnection.createDataChannel('' + label, conf)
+        const dataChannel = this.#handleDataChannel(channel, opts)
+    
+        if (this.#announceNegotiation == null) {
+            this.#announceNegotiation = false
+            this.dispatchEvent(new Event('negotiationneeded'))
+        }
+    
+        return dataChannel
     }
 
-    createOffer(): Promise<globalThis.RTCSessionDescriptionInit | any> {
+    createOffer(): Promise<globalThis.RTCSessionDescriptionInit> & Promise<void> {
```

Comment:
there are some annoying deprecated overloads for some of these methods, we shouldn't support them